libs/etterlatte-database/src/main/kotlin/JdbcUtils.kt

```diff
@@ -51,3 +54,189 @@ inline fun <reified T : Any> PreparedStatement.setJsonb(
     this.setObject(parameterIndex, jsonObject)
     return this
 }
+
+fun <T> ConnectionAutoclosing.hent(
+    statement: String,
+    params: List<SQLParameter>,
+    modus: Uthentingsmodus = Uthentingsmodus.SINGLE_OR_NULL,
+    konverterer: (ResultSet.(Any?) -> T),
+): T? =
+    hentConnection {
+        with(it) {
+            val stmt = prepareStatement(statement.trimMargin())
+            params.forEachIndexed { index, param -> param.settParameter(index + 1, stmt) }
```

Comment:
Jeg føler at å abstrahere bort parameterindeks med indeks i en liste kan føre til en del utilsiktede feil -- når jeg ser setParam(3, "wow") er betydningen av 3 i forhold til spørringen eksplisitt, mens her blir koblingen implisitt basert på rekkefølge i en liste. Til sammenligning foretrekker jeg mer noe ala kotliquery -- det er en mer eksplisitt sammenheng mellom SQL-spørringen og parametrene jeg sender inn. 

Med en liste-håndtering vil jeg se for meg flere feil knyttet til ombytting av parametre / feil når man legger til ekstra parameter i listen. Gjenbruk av samme verdi flere steder i SQL-spørringen er også litt vanskeligere å se sammenhenger i.

Hadde ikke en bedre vei vært å bare erstatte den egne bruken av jdbc med kotliquery?

Reply:
Ja, denne PR-en er på eit vis eit svar på spørsmålet _skal vi bruke kotliquery eller ikkje?_ Fram til no har vi hatt ein slags hybrid kor vi bruker det i nokre appar, men ikkje i alle, og så får jo den modellen seg ein trøkk no når @sebassonav held på å flytte vilkårsvurdering, som bruker kotliquery, inn i behandling, som ikkje bruker kotliquery.

Så da er det vel no eit tidspunkt for å ta valet om vi skal køyre kotliquery eller ikkje. Viss vi ikkje skal det, har eg veldig lyst på eit abstraksjonslag a la dette, for å skilje teknisk knotete frå funksjonelt. Viss vi skal halde oss til kotliquery, er det berre å skrape denne PR-en.

Etter å ha sett at det er éin person som i praksis vedlikeheld kotliquery, har eg vorte langt meir skeptisk til at vi skal basere oss på det. Men det er vel også eit tema for utviklarsynk etterpå.

Reply:
Til biten med å abstrahere bort indeks: Gode poeng! Det kan vi godt endre på. Ser for meg at vi kan utvide SQLObject, SQLLong osv alle med indeks som felt.